### ISSUE TITLE
only one epoch is ok

### ISSUE THREAD

**Issue by Miraclemarvel55** [noahchalifour/baidu-deepspeech2#2]

**Comment by Miraclemarvel55** [comment]
after replace this function,issue will solved. 将下面的函数替换,问题将会解决.
#utils.py
def create_data_generator(directory, max_input_length, max_label_length, batch_size=64):

    x, y, input_lengths, label_lengths = [], [], [], []
    while True:
        try:
            with open(os.path.join(directory, 'metadata.csv'), 'r') as metadata:
                metadata_reader = csv.DictReader(metadata, fieldnames=['filename', 'spec_length', 'labels_length', 'labels'])
                next(metadata_reader)
                for row in metadata_reader:
                    audio = np.load(os.path.join(directory, row['filename'] + '.npy'))
                    x.append(audio)
                    y.append([int(i) for i in row['labels'].split(' ')])
                    input_lengths.append(row['spec_length'])
                    label_lengths.append(row['labels_length'])
                    if len(x) == batch_size:
                        yield {
                            'inputs': tf.keras.preprocessing.sequence.pad_sequences(x, maxlen=max_input_length, padding='post'),
                            'labels': tf.keras.preprocessing.sequence.pad_sequences(y, maxlen=max_label_length, padding='post'),
                            'input_lengths': np.asarray(input_lengths),
                            'label_lengths': np.asarray(label_lengths)
                        }, {
                            'ctc': np.zeros([batch_size])
                        }
                        x, y, input_lengths, label_lengths = [], [], [], []
        except:
            pass
